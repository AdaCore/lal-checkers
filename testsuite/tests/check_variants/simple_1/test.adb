procedure Test is
   type My_Variant(x : Integer) is record
      case x is
         when 0 .. 500 =>
            case x is
               when 0 .. 250 =>
                  z : Integer;
               when others =>
                  null;
            end case;
         when others =>
            y : Boolean;
      end case;
   end record;
   x : My_Variant := (x => 600, y => True);
   res : Integer;
begin
   res := x.z;
end Ex1;
