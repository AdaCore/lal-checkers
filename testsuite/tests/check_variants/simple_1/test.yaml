driver: python
helper: test_field_exists.py
