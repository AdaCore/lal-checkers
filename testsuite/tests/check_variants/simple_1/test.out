{
  "always": [
    {
      "precise": true, 
      "trace": [
        "assign0", 
        "assign1", 
        "assign2", 
        "assume0", 
        "start0"
      ]
    }
  ], 
  "le_t_eq_v": [
    {
      "precise": true, 
      "trace": [
        "assign0", 
        "assign1", 
        "assign2", 
        "assume0", 
        "start0"
      ]
    }
  ]
}
